### PR TITLE
Added support for TntId setter API

### PR DESCRIPTION
**Pull request by swarna04**
## Description

[Ref: MOB-16757]

- Added support for an API to set tntId in the SDK
**func setTntId(_ id: String?)**
- Added functionality to derive Edge host value from the tntId location hint.
- Added unit, functional and integration tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by sbenedicadb — APPROVED** [review]
looks good with a couple of nit items.  

curious why we are storing "empty" `tntId` as empty string, but "empty" `edgeHost` as nil

File: AEPTarget/Sources/Target+PublicAPI.swift [hunk 1]
```diff
@@ -226,6 +226,20 @@ import Foundation
...
 
+    /// Sets the Test and Target user identifier.
+    /// The provided Tnt Id is persisted in the SDK and attached to all subsequent Target requests. It is used to
+    /// derive the edge host value in the SDK, which is also persisted and used in future Target requests.
+    ///
+    /// This ID is preserved between app upgrades, is saved and restored during the standard application backup process,
+    /// and is removed at uninstall, upon privacy status update to opt out or when AEPTarget.resetExperience is called.
+    ///
+    /// - Parameter id: a string containing the value of the Tnt Id to be set in the SDK.
+    static func setTntId(_ id: String?) {
```
[the remark this answers is not in the document]
Comment (swarna04) [reply AEPTarget/Sources/Target+PublicAPI.swift:240]: It is visible in objc, I've updated the objc testapp as well.

File: AEPTarget/Sources/Target.swift [hunk 2]
```diff
@@ -730,22 +735,53 @@ public class Target: NSObject, Extension {
...
         if targetState.privacyStatusIsOptOut, let tntId = tntId, !tntId.isEmpty {
-            Log.debug(label: Target.LOG_TAG, "setTntId - Cannot update Target tntId due to opt out privacy status.")
+            Log.debug(label: Target.LOG_TAG, "setTntIdInternal - Cannot update Target tntId due to opt out privacy status.")
             return
         }
 
-        if tntIdValuesAreEqual(newTntId: tntId, oldTntId: targetState.tntId) {
-            Log.debug(label: Target.LOG_TAG, "setTntId - New tntId value is same as the existing tntId \(String(describing: targetState.tntId)).")
+        if tntId == targetState.tntId {
+            Log.debug(label: Target.LOG_TAG, "setTntIdInternal - New tntId value is same as the existing tntId \(String(describing: tntId)).")
```
[the remark this answers is not in the document]
Comment (swarna04) [reply AEPTarget/Sources/Target.swift]: I believe we need some of these as debug logs to interpret the different scenarios. Added a trace log as well.

File: AEPTarget/Sources/Target+PublicAPI.swift [hunk 3]
```diff
@@ -226,10 +226,13 @@ import Foundation
         }
     }
 
-    /// Sets the Test and Target user identifier.
+    /// Sets the Target user identifier.
+    ///
     /// The provided Tnt Id is persisted in the SDK and attached to all subsequent Target requests. It is used to
     /// derive the edge host value in the SDK, which is also persisted and used in future Target requests.
     ///
+    /// If the provided Tnt Id is nil or empty, the SDK will remove the Tnt Id and edge host values from persistence.
```
Comment (sbenedicadb) [inline AEPTarget/Sources/Target+PublicAPI.swift:236]: 👍

**Review by sbenedicadb — APPROVED** [review]

**Pull request merged**